addons/website/static/src/interactions/header/base_header.js

```diff
@@ -0,0 +1,209 @@
+        },
+    }
+
+    //--------------------------------------------------------------
+    // Life Cycle
+    //--------------------------------------------------------------
+
+    setup() {
+        this.topGap = 0;
+        this.atTop = false;
+
+        this.cssAffixed = false;
+        this.bodyNoScroll = false;
+
+        this.transitionCount = 0;
+        this.transitionActive = true;
+
+        this.isVisible = true;
+        this.isScrolled = false;
+        this.hasScrolled = false;
+        this.closeDropdowns = false;
+
+        this.isOverlay = !!this.el.closest(".o_header_overlay, .o_header_overlay_theme")
+
+        this.mainEl = this.el.parentElement.querySelector("main");
+        this.hideEl = this.el.querySelector(".o_header_hide_on_scroll");
+    }
+
+    //--------------------------------------------------------------
+    // Event Handlers
+    //--------------------------------------------------------------
+
+    disableScroll() {
+        if (uiUtils.getSize() < SIZES.LG) {
+            this.bodyNoScroll = true;
+        }
+    };
+
+    enableScroll() {
+        this.bodyNoScroll = false;
+    };
+
+    onScroll() {
+
+        const scroll = document.scrollingElement.scrollTop;
+
+        if (!this.hasScrolled) {
+            this.hasScrolled = true;
+            if (scroll > 0) {
+                this.adjustPosition();
+            }
+        } else {
+            this.closeDropdowns = true;
+        }
+
+        if (this.closeDropdowns) {
+            const dropdownToggleEls = this.el.querySelectorAll(".dropdown-toggle.show");
+            for (const dropdownToggleEl of dropdownToggleEls) {
+                Dropdown.getOrCreateInstance(dropdownToggleEl).hide();
+            }
+        }
+    };
+
+    onResize() {
+        this.adjustScrollbar();
+        if (
+            document.body.classList.contains('overflow-hidden')
+            && uiUtils.getSize() >= SIZES.LG
+        ) {
+            const offCanvasEls = this.el.querySelectorAll(".offcanvas.show");
+            for (const offCanvasEl of offCanvasEls) {
+                Offcanvas.getOrCreateInstance(offCanvasEl).hide();
+            }
+            // Compatibility: can probably be removed, there is no such elements in
+            // default navbars... although it could be used by custo.
+            const collapseEls = this.el.querySelectorAll(".navbar-collapse.show");
+            for (const collapseEl of collapseEls) {
+                Collapse.getOrCreateInstance(collapseEl).hide();
+            }
+        }
+    };
+
+    //--------------------------------------------------------------
+    // Animation Handlers
+    //--------------------------------------------------------------
+
+    adaptToHeaderChange() {
+        for (const callback of extraMenuUpdateCallbacks) {
+            callback();
+        }
+        this.adjustMainPadding();
+    }
+
+    adaptToHeaderChangeLoop(addCount = 0) {
+        this.adaptToHeaderChange();
```

Comment:
note that both the settimeout and the request animation frame are asynchronous => so the interaction could have been destroyed in the meantime. for the setTimeout, you could instead have used the `this.waitForTimeout` function, but since there is no similar function for animation frame, i guess you can simply add a check here:
```js
if (this.isDestroyed) {
  return;
}
```

Reply:
i mean, on the first line of the function of course